### PR TITLE
zoneminder, nixos/zoneminder: 1.36.28 -> 1.36.31, fix issues relating to upgrade

### PR DESCRIPTION
**Pull request by peat-psuwit**
###### Description of changes

Changelog for [1.36.29](https://forums.zoneminder.com/viewtopic.php?t=32104) and [1.36.30,1.36.31](https://forums.zoneminder.com/viewtopic.php?t=32122). The nixos module is updated to make it runs correctly after an upgrade.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable: (tests run on aarch64-linux)
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/zoneminder.nix [hunk 1]
```diff
@@ -326,6 +327,15 @@ in {
...
           ${zoneminder}/bin/zmupdate.pl -nointeractive
+          ${zoneminder}/bin/zmupdate.pl --nointeractive -f
+
+          # Update ZM's Nix store path in the configuration table. Do nothing if the config doesn't
+          # contain ZM's Nix store path.
+          ${config.services.mysql.package}/bin/mysql -u zoneminder zm << EOF
+            UPDATE Config
+              SET Value = REGEXP_REPLACE(Value, "^/nix/store/[^-/]+-zoneminder-[^/]+", "${pkgs.zoneminder}")
+              WHERE Name = "ZM_FONT_FILE_LOCATION";
+          EOF
```
Comment (peat-psuwit) [inline nixos/modules/services/misc/zoneminder.nix:338]: This code assumes that the Nix store is at `/nix/store`. Is there a variable or something which I can use to determine the path to Nix store here?

**Comment by peat-psuwit** [comment]
At the moment the resulting binary seems to be missing a runpath for libmariadb?

```
$ ldd /nix/store/7404x3bgi3ig6vs8y06pmsgg6ph2cjjl-zoneminder-1.36.32/bin/zmc | grep 'not found'
        libmariadb.so.3 => not found
```

And this seems to happen when I upgrade my machine to 22.11. Looking into this issue...

**Comment by peat-psuwit** [comment]
The above issue is actually fixed by #203369, backported to 22.11 as #209893. So that's no issue now.

Ping the last contributors: @SuperSandro2000 @risicle @FliegendeWurst

File: nixos/modules/services/misc/zoneminder.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/services/misc/zoneminder.nix:338]: If that is changed, you need to recompile everything and it is hardcoded in a lot of places.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**